drivers/scheduler/k8s/k8s.go

```diff
@@ -616,6 +616,72 @@ func (k *k8s) GetNodesForApp(ctx *scheduler.Context) ([]node.Node, error) {
 	return result, nil
 }
 
+func (k *k8s) GenerateSupportBundle(ctx *scheduler.Context) (string, error) {
+	var output string
+	var err error
+	for _, core := range ctx.App.SpecList {
+		if obj, ok := core.(*apps_api.Deployment); ok {
+			output += insertLineBreak(obj.Name)
+			var depStatus apps_api.DeploymentStatus
+			if depStatus, err = k8sutils.DescribeDeployment(obj); err != nil {
+				output += fmt.Sprintf("%v", &ErrFailedToValidateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to get status of deployment: %v. Err: %v", obj.Name, err),
+				})
+			}
+			//Dump depStatus
+			output += fmt.Sprintf("%v\n", depStatus)
+			pods, _ := k8sutils.GetDeploymentPods(obj)
+			for _, pod := range pods {
+				output += insertLineBreak(pod.Name)
+				output += fmt.Sprintf("%v\n", pod.Status)
+				for _, conStat := range pod.Status.ContainerStatuses {
+					output += insertLineBreak(conStat.Name)
+					output += fmt.Sprintf("%v\n", conStat)
+					output += insertLineBreak("END Container")
+				}
+				output += insertLineBreak("END Pod")
+			}
+			output += insertLineBreak("END Deployment")
+		} else if obj, ok := core.(*apps_api.StatefulSet); ok {
+			output += insertLineBreak(obj.Name)
+			var ssetStatus apps_api.StatefulSetStatus
+			if ssetStatus, err = k8sutils.DescribeStatefulSet(obj); err != nil {
+				output += fmt.Sprintf("%v", &ErrFailedToValidateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to get status of statefulset: %v. Err: %v", obj.Name, err),
+				})
+			}
+			//Dump ssetStatus
+			output += fmt.Sprintf("%v\n", ssetStatus)
+			pods, _ := k8sutils.GetStatefulSetPods(obj)
+			for _, pod := range pods {
+				output += insertLineBreak(pod.Name)
+				output += fmt.Sprintf("%v\n", pod.Status)
+				for _, conStat := range pod.Status.ContainerStatuses {
+					output += insertLineBreak(conStat.Name)
+					output += fmt.Sprintf("%v\n", conStat)
+					output += insertLineBreak("END Container")
+				}
+				output += insertLineBreak("END Pod")
+			}
+			output += insertLineBreak("END StatefulSet")
+		} else {
+			output += insertLineBreak("UNKNOWN/UNSUPPORTED OBJ TYPE")
```

Comment:
you will have to remove the else block because now we are parsing the entire list of specs which include storage specs. They are not from the above list but still valid specs. We can rely on the parser that it is not going to let unsupported objects in the scheduler.

Reply:
We have the entire list but we don't have the handling for all kinds yet. That handling means more of the APIs for those kinds need to be developed and integrated. That should be a separate task.

Reply:
What @piyush-nimbalkar means to say is `SpecList` will contain all sorts of spec (including `PersistentVolumeClaim`, `StorageClass`. This change was made recently. So your current code, will cause that to come to this else case and add the unsupported type message. So remove this else block.